### PR TITLE
Ability to configure global options

### PR DESCRIPTION
**Pull request by robin-thoni**
Issue #11

#### Bugfixes

* Fixed error when input are not in a form (cf2b00f)
* Fixed date object error (7417a93)

#### Features

* Added service to configure settings globally
* Added demo

### CODE REVIEW

File: src/components/mdpDatePicker/mdpDatePicker.js [hunk 1]
```diff
@@ -423,15 +427,15 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
                 };
 
                 ngModel.$validators.minDate = function(modelValue, viewValue) {
-                    return minDateValidator(viewValue, scope.dateFormat, scope.minDate);
+                    return minDateValidator(viewValue, scope.dateFormat, minDate);
```
Comment (dpoetzsch) [inline src/components/mdpDatePicker/mdpDatePicker.js]: I think, here you break the possibility to change the `minDate` after creation of the picker using the scope binding. I believe we have to do a `scope.minDate || $mdpLocale.date.minDate` here (and cleanup the constant definition above).

File: src/components/mdpDatePicker/mdpDatePicker.js [hunk 2]
```diff
@@ -423,15 +427,15 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
...
                 };
 
                 ngModel.$validators.maxDate = function(modelValue, viewValue) {
-                    return maxDateValidator(viewValue, scope.dateFormat, scope.maxDate);
+                    return maxDateValidator(viewValue, scope.dateFormat, maxDate);
```
Comment (dpoetzsch) [inline src/components/mdpDatePicker/mdpDatePicker.js]: same as with `minDate`

File: src/components/mdpDatePicker/mdpDatePicker.js [hunk 3]
```diff
@@ -423,15 +427,15 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
...
                 };
 
                 ngModel.$validators.filter = function(modelValue, viewValue) {
-                    return filterValidator(viewValue, scope.dateFormat, scope.dateFilter);
+                    return filterValidator(viewValue, scope.dateFormat, dateFilter);
```
Comment (dpoetzsch) [inline src/components/mdpDatePicker/mdpDatePicker.js]: same as with `minDate`

File: src/components/mdpDatePicker/mdpDatePicker.js [hunk 4]
```diff
@@ -484,9 +488,9 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
 
                 scope.showPicker = function(ev) {
                     $mdpDatePicker(ngModel.$modelValue, {
-                        minDate: scope.minDate,
-                        maxDate: scope.maxDate,
-                        dateFilter: scope.dateFilter,
+                        minDate: minDate,
+                        maxDate: maxDate,
+                        dateFilter: dateFilter,
```
Comment (dpoetzsch) [inline src/components/mdpDatePicker/mdpDatePicker.js]: same as with `minDate` above (for all three lines)

File: src/components/mdpTimePicker/mdpTimePicker.js [hunk 5]
```diff
@@ -250,12 +250,12 @@ module.provider("$mdpTimePicker", function() {
                 locals: {
                     time: time,
                     autoSwitch: options.autoSwitch,
-                    ampm: options.ampm
+                    ampm: options.ampm == null ? $mdpLocale.time.ampm : options.ampm
```
Comment (dpoetzsch) [inline src/components/mdpTimePicker/mdpTimePicker.js]: I think we might go better with `angular.isDefined(options.ampm) ? options.ampm : $mdpLocale.time.ampm` here

File: src/components/mdpTimePicker/mdpTimePicker.js [hunk 6]
```diff
@@ -369,11 +372,11 @@ module.directive("mdpTimePicker", ["$mdpTimePicker", "$timeout", function($mdpTi
             };
             
             ngModel.$validators.minTime = function(modelValue, viewValue) {
-                return minTimeValidator(viewValue, scope.timeFormat, scope.minTime);
+                return minTimeValidator(viewValue, scope.timeFormat, minTime);
```
Comment (dpoetzsch) [inline src/components/mdpTimePicker/mdpTimePicker.js]: same as with `minDate` in date picker

File: src/components/mdpTimePicker/mdpTimePicker.js [hunk 7]
```diff
@@ -369,11 +372,11 @@ module.directive("mdpTimePicker", ["$mdpTimePicker", "$timeout", function($mdpTi
...
             };
 
             ngModel.$validators.maxTime = function(modelValue, viewValue) {
-                return maxTimeValidator(viewValue, scope.timeFormat, scope.maxTime);
+                return maxTimeValidator(viewValue, scope.timeFormat, maxTime);
```
Comment (dpoetzsch) [inline src/components/mdpTimePicker/mdpTimePicker.js]: same as with `minDate` in date picker

**Review by dpoetzsch — CHANGES_REQUESTED** [review]
Lovely, thanks!

I added some remarks in the code. Only little issues though.

Furthermore, it looks like all the files in demo-dist are auto-generated? If yes, we should exclude them from the repository.

Also, at some point we should merge your demo with the original one (which might be broken at the moment)... But this is work for another PR.

**Comment by robin-thoni** [comment]
> I added some remarks in the code. Only little issues though.

I fixed them

> Furthermore, it looks like all the files in demo-dist are auto-generated? If yes, we should exclude them from the repository.

I was hesitant about that before making the PR, and I ended by keeping it. Now, it is ignored.

> Also, at some point we should merge your demo with the original one (which might be broken at the moment)... But this is work for another PR.

Yes :) I spotted a few other bugs, I'll probably fix them

**Pull request closed without merging**